pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -211,6 +212,49 @@ func GetMachineTypesFromCloudProfile(cloudProvider gardenv1beta1.CloudProvider,
 	return machineTypes
 }
 
+// GetMachineImagesFromCloudProfile retrieves a list of machine images from the cloud profile
+func GetMachineImagesFromCloudProfile(profile *gardenv1beta1.CloudProfile) ([]gardenv1beta1.MachineImage, error) {
+	cloudProvider, err := DetermineCloudProviderInProfile(profile.Spec)
+	if err != nil {
+		return nil, err
+	}
+	switch cloudProvider {
+	case gardenv1beta1.CloudProviderAWS:
+		return profile.Spec.AWS.Constraints.MachineImages, nil
+	case gardenv1beta1.CloudProviderAzure:
+		return profile.Spec.Azure.Constraints.MachineImages, nil
+	case gardenv1beta1.CloudProviderGCP:
+		return profile.Spec.GCP.Constraints.MachineImages, nil
+	case gardenv1beta1.CloudProviderPacket:
+		return profile.Spec.Packet.Constraints.MachineImages, nil
+	case gardenv1beta1.CloudProviderOpenStack:
+		return profile.Spec.OpenStack.Constraints.MachineImages, nil
+	}
+	return nil, nil
+}
+
+// SetMachineImagesForCloudProvider sets the machine images for the cloud provider
+func SetMachineImagesForCloudProvider(profile *gardenv1beta1.CloudProfile, images []gardenv1beta1.MachineImage) error {
```

Comment:
```suggestion
func SetMachineImages(profile *gardenv1beta1.CloudProfile, images []gardenv1beta1.MachineImage) error {
```

the cloud provider parameter is no longer part of the signature